src/components/CloudHeader/CloudHeader.js

```diff
@@ -0,0 +1,164 @@
+    document.body.appendChild(this.portalNode);
+  }
+
+  componentWillUnmount() {
+    this.portalNode.parentNode.removeChild(this.portalNode);
+  }
+
+  render() {
+    const {
+      className,
+      companyName,
+      productName,
+      logoHref,
+      renderMenu,
+      renderLogo,
+      renderSearch,
+      renderNotification,
+      renderApplication,
+      renderUser,
+      links,
+      ...other
+    } = this.props;
+
+    const searchIcon = (
+      <svg viewBox="0 0 16 16" fillRule="evenodd">
+        <title>search</title>
+        <path d="M6 2c2.2 0 4 1.8 4 4s-1.8 4-4 4-4-1.8-4-4 1.8-4 4-4zm0-2C2.7 0 0 2.7 0 6s2.7 6 6 6 6-2.7 6-6-2.7-6-6-6zm10 13.8L13.8 16l-3.6-3.6 2.2-2.2z" />
+        <path d="M16 13.8L13.8 16l-3.6-3.6 2.2-2.2z" />
+      </svg>
+    );
+
+    const notificationIcon = (
+      <svg viewBox="0 0 16 16">
+        <title>notifications</title>
+        <path d="M9 1.11V0H7v1.11A5.022 5.022 0 0 0 3.1 4.9L1 14h5a2 2 0 0 0 4 0h5l-2.1-9.1A5.022 5.022 0 0 0 9 1.11z" />
+      </svg>
+    );
+
+    const applicationIcon = (
+      <svg viewBox="0 0 16 16">
+        <title>applications</title>
+        <circle cx="2" cy="2" r="2" />
+        <circle cx="8" cy="2" r="2" />
+        <circle cx="14" cy="2" r="2" />
+        <circle cx="2" cy="8" r="2" />
+        <circle cx="8" cy="8" r="2" />
+        <circle cx="14" cy="8" r="2" />
+        <circle cx="2" cy="14" r="2" />
+        <circle cx="8" cy="14" r="2" />
+        <circle cx="14" cy="14" r="2" />
+      </svg>
+    );
+
+    const userIcon = (
+      <svg className="bx--cloud-header__user-icon" viewBox="0 0 24 24">
+        <title>user</title>
+        <path d="M12 0C5.4 0 0 5.4 0 12s5.4 12 12 12 12-5.4 12-12S18.6 0 12 0zm0 4.6c2.8 0 5 2.2 5 5s-2.2 5-5 5-5-2.2-5-5 2.2-5 5-5zm7 14.5c-1.8 1.8-4.3 2.9-7 2.9s-5.2-1.1-7-2.9v-1.6c0-1.3.7-2 2-2h10c1.3 0 2 .7 2 2v1.6z" />
+      </svg>
+    );
+
+    const { isMenuOpen } = this.state;
+    const cloudHeaderClasses = classNames('bx--cloud-header', className);
+
+    return (
+      <nav key="nav" className={cloudHeaderClasses} {...other}>
+        <CloudHeaderWrapper>
+          {renderMenu && <CloudHeaderMenu onClick={this.handleMenuClick} />}
+          <CloudHeaderLogo
+            className={!renderMenu ? 'bx--cloud-header-brand--no-menu' : null}
+            companyName={companyName}
+            productName={productName}
+            href={logoHref}>
+            {renderLogo && renderLogo()}
+          </CloudHeaderLogo>
+          <CloudHeaderList>
+            {links &&
+              links.map((link, i) => {
+                return (
+                  <CloudHeaderListItem key={i} href={link.href}>
+                    {link.linkText}
+                  </CloudHeaderListItem>
+                );
+              })}
+          </CloudHeaderList>
+        </CloudHeaderWrapper>
+        <CloudHeaderWrapper>
+          <CloudHeaderList>
+            {renderSearch && (
```

Comment:
Should these dropdowns be rendering inside of our markup code so that people won't have to absolutely position whatever they decide to render from `renderSearch`, `renderAccount`, etc?

Reply:
Possibly, but for a V1 and until we get more dev use cases I think this is fine

Reply:
Since we have some people requesting this, I think we should at least make sure that adding in a custom dropdown for account is easy. What do you think?

Reply:
How should we go about changing this then?

Reply:
It should be just figuring out where that UI portion should sit in the DOM, like search might come before the list, account might just be in the same node, and then calling the `render*` function to generate the markup needed, potentially?

Reply:
Can you give a more concrete example of what needs to be changed?

Reply:
Wouldn't it just be something like the following:

```js
        <CloudHeaderWrapper>
          {renderSearch && renderSearch()}
          <CloudHeaderList>
            {renderSearch && (
              <CloudHeaderListItem
                onClick={this.handleIconClick}
                data-icon="Search"
                isIcon>
                {searchIcon}
              </CloudHeaderListItem>
            )}
            {renderNotification && (
              <CloudHeaderListItem
                onClick={this.handleIconClick}
                data-icon="Notification"
                isIcon>
                {notificationIcon}
                {renderNotification()}
              </CloudHeaderListItem>
            )}
            {renderApplication && (
              <CloudHeaderListItem
                onClick={this.handleIconClick}
                data-icon="Application"
                isIcon>
                {applicationIcon}
                {renderApplication()}
              </CloudHeaderListItem>
            )}
            {renderUser && (
              <CloudHeaderListItem
                onClick={this.handleIconClick}
                data-icon="User"
                isIcon>
                {userIcon}
                {renderUser()}
              </CloudHeaderListItem>
            )}
          </CloudHeaderList>
        </CloudHeaderWrapper>
```

Reply:
Not sure I follow, when I add `{renderSearch && renderSearch()}` below `CloudHeaderWrapper`, it calls render Search every time any icon is clicked?